src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,304 @@
+
+    @ParameterizedTest
+    @MethodSource("createCorrectWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsCorrect(String correctWord) {
+        assertTrue(correctWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsWrong(String wrongWord) {
+        assertFalse(wrongWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    private static List<String> createCorrectEmails() {
+        List<String> correctEmails = new ArrayList<>();
+        correctEmails.add("pushkin@ukr.net");
+        correctEmails.add("pushkin.145@gmail.com");
+        correctEmails.add("pushkin@mail.ru");
+        correctEmails.add("pushkin.pushkin@mail.ru");
+        correctEmails.add("pushkin.pushkin@mail.mail.ru");
+        correctEmails.add("1@ukr.net");
+        correctEmails.add("1@111.ru");
+
+        return correctEmails;
+    }
+
+    private static List<String> createWrongEmails() {
+        List<String> wrongEmails = new ArrayList<>();
+        wrongEmails.add("pushkin@ukr.n");
+        wrongEmails.add(".145@gmail.com");
+        wrongEmails.add("pushkin@@mail.ru");
+        wrongEmails.add("1@111.11");
+        wrongEmails.add("pushkin.11");
+        wrongEmails.add("pushkin.1u");
+        wrongEmails.add("pushkin.u1");
+        wrongEmails.add("1@111.1u");
+        wrongEmails.add("1@111.1ru");
+        wrongEmails.add("1@111.ru1");
+        wrongEmails.add("1@ma~il.ru");
+
+        List<Character> correctCharacters = List.of('.', '_', '+', '-');
+
+        wrongEmails.addAll(getStringsWithSpecialSymbol("pu", "shkin@ukr.net", correctCharacters));
+        wrongEmails.addAll(getStringsWithRussianOrUkrCharacter("pu", "shkin@ukr.net"));
+
+        return wrongEmails;
+    }
+
+    private static List<String> createCorrectPasswords() {
+        List<String> correctPasswords = new ArrayList<>();
+        correctPasswords.add("aA1");
+        correctPasswords.add("1aA");
+        correctPasswords.add("a1A");
+        correctPasswords.add("#aA1");
+        correctPasswords.add("aA1@");
+        correctPasswords.add("~aA1`");
+        correctPasswords.add("~`aA1`!@#$%^&*()_+-=");
+
+        correctPasswords.addAll(getStringsWithSpecialSymbol("a", "A1", Collections.emptyList()));
+
+        return correctPasswords;
+    }
+
+    private static List<String> createWrongPasswords() {
+        List<String> wrongPasswords = new ArrayList<>();
+        wrongPasswords.add("");
+        wrongPasswords.add("a");
+        wrongPasswords.add("1");
+        wrongPasswords.add("A");
+        wrongPasswords.add("a1");
+        wrongPasswords.add("aA");
+        wrongPasswords.add("A1");
+        wrongPasswords.add("aa1");
+        wrongPasswords.add("a11");
+        wrongPasswords.add("AA1");
+
+        wrongPasswords.add(" aA1");
+        wrongPasswords.add("aA1 ");
+        wrongPasswords.add(" aA1 ");
+        wrongPasswords.add("a A1");
+
```

Comment:
Why do we need this empty line?

Reply:
Just for convenience at the time of writing. I forgot to delete it. Thanks.